### ISSUE TITLE
bug(deps): dependency issues when opting into tests with upgraded generator

### ISSUE THREAD

**Issue by dustingraves** [prescottprue/generator-react-firebase#101]
**Do you want to request a *feature* or report a *bug*?**  
Report a bug

**What is the current behavior?**
After a fresh install of the generator, there are package dependency issues.
```npm WARN redux-form@7.3.0 requires a peer of redux@^3.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN chai-enzyme@0.6.1 requires a peer of react@^0.14.0 || ^15.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN chai-enzyme@0.6.1 requires a peer of react-dom@^0.14.0 || ^15.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme@2.9.1 requires a peer of react@0.13.x || 0.14.x || ^15.0.0-0 || 15.x but none is installed. You must install peer dependencies yourself.
npm WARN react-addons-test-utils@15.6.2 requires a peer of react-dom@^15.4.2 but none is installed. You must install peer dependencies yourself.
npm WARN react-test-renderer@15.6.2 requires a peer of react@^15.6.2 but none is installed. You must install peer dependencies yourself.
npm WARN react-element-to-jsx-string@5.0.7 requires a peer of react@^0.14.8 || ^15.0.1 but none is installed. You must install peer dependencies yourself.
```
If you attempt to run `npm test` you get 9 `Module not found` errors relating to the dependency issues

**What is the expected behavior?**
It should have the required dependencies in package.json and not error when installing & testing

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**
`npm test` worked before the upgrade to react@16 and @material-ui/core@v1.0.0

**Comment by prescottprue** [comment]
@dustingraves How do you feel about Jest? I started moving the tests over to that on the `jest` branch for snapshot testing support along with easier dependency management.

I think it may be easier to switch over than figure out all of the correct updates without breaking things, but not totally sure.

**Comment by dustingraves** [comment]
I haven't used `Jest`, but it looks pretty good. I was able to fix all of the dependencies. It only took me a few hours of googling...

### tests/test-bundler.js
```javascript
import 'core-js/es6/map'
import 'core-js/es6/set'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// ...
// Enzyme React 16 adapter
configure({ adapter: new Adapter() })
```

### package.json
```javscript
// add packages
"chai-enzyme": "^1.0.0-beta.0",
"core-js": "^2.5.7",
"enzyme": "^3.3.0",
"enzyme-adapter-react-16": "^1.1.1",
"react-test-renderer": "^16.4.0",

// remove package
"react-addons-test-utils": "^15.5.1",
```

### tests/layouts/CoreLayout.spec.js
```javascript
import React from 'react'
// import TestUtils from 'react-addons-test-utils'
import ShallowRenderer from 'react-test-renderer/shallow'
import CoreLayout from 'layouts/CoreLayout/CoreLayout'

function shallowRender(component) {
const renderer = new ShallowRenderer()
```
The only straggler is redux-form.

**Comment by dustingraves** [comment]
The alternative to the `core-js` implementation was a `babel-polyfill` and reference it in the files import of the karma config.

**Comment by prescottprue** [comment]
Well, since you were able to fix it so quick, we can just go that route. Thanks for posting all the info!

**Comment by prescottprue** [comment]
Based on your input I was able to make changes that fix the tests (will go out in `v2.0.1`). Something else to note: tests are not included by default as of `2.0.0`, but they are a feature option. If you want to include tests or their dependencies, make sure to select this option.
